### PR TITLE
Issue 11854 add publish and expire date fields

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/contenttype/business/FieldFactoryImpl.java [hunk 1]
```diff
@@ -534,25 +535,24 @@ public String nextAvailableColumn(Field field) throws DotDataException {
       return field.dataType().toString();
     }
 
-
-    String dataType = field.dataType().toString();
+    String dbColumn = field.dbColumn().replaceAll("[0-9]", "");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldFactoryImpl.java]: this should be on the UtilsMethod or some escape utility

Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/FieldFactoryImpl.java]: set to final

**Comment by wezell** [comment]
I think the data we are trying to get at is available in the Field Impl class, meaning we have:

- com.dotcms.contenttype.model.field.DateField
- com.dotcms.contenttype.model.field.DateTimeField
- com.dotcms.contenttype.model.field.TimeField

**Comment by wezell** [comment]
I think we need to work on this using the data available via the REST endpoints

**Pull request closed without merging**